rewrite-java/src/main/java/org/openrewrite/java/marker/JavaSourceSet.java

```diff
@@ -330,7 +330,8 @@ private static String entryNameToClassName(String entryName) {
     private static boolean isDeclarable(String className) {
         int dotIndex = Math.max(className.lastIndexOf("."), className.lastIndexOf('$'));
         className = className.substring(dotIndex + 1);
-        char firstChar = className.charAt(0);
-        return Character.isJavaIdentifierPart(firstChar) && !Character.isDigit(firstChar);
+        return !className.isEmpty() &&
```

Comment:
should we maybe update/add a test for this?

Reply:
Hadn't wanted to add a dependency to test a one line change, but added it just now so you can see if we want to keep this in.

Reply:
There are other, smaller/more direct ways to test this I think

Reply:
Changed visibility and now testing with just the class name; definitely lightweight to test now, and the test would serve mostly as documentation and safeguard against regression.